server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/rfc/model/RequestObject.scala

```diff
@@ -0,0 +1,43 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.jmap.rfc.model
+
+object RequestObject {
+
+  import play.api.libs.json._
+
+  case class Capability(value: String) extends AnyVal
+  case class ClientId(value: String) extends AnyVal
```

Comment:
not clientId, createdId, and it's just name, not type. (The true type is Id) (https://jmap.io/spec-core.html#the-request-object - createdIds item)

Reply:
Keep a clientId types that wraps Id class as AnyVal. Strong typing ;-)

Reply:
It can't be: see [Summary of Limitations](https://docs.scala-lang.org/overviews/core/value-classes.html)

> … must have only a primary constructor with exactly one public, val parameter whose type is not a user-defined value class. (From Scala 2.11.0, the parameter may be non-public.)

you can only put primitive types inside a Value Class. User-defined type are not accepted

Reply:
Don't overrate Value Class. We probably prefer not using Value Class than loosing expressiveness.